Update macOS VMs in CI/CD to v15 on Intel arch (#791)
ansible/pylibssh: docs/changelog-fragments/791.contrib.rst

Reviewer: @komaldesai13 it's best to accept the suggested changes as is and preview what it looks like. Here's a PR preview: https://ansible-pylibssh--791.org.readthedocs.build/en/791/changelog/#changelog. With your rephrasing, the rendering is broken since RST merges the lines. Plus there's no periods in sentences.

We should be careful to write change notes that follow the same style. These previews I've set up will help you with judging that. The goal is to have consistent style across the change log. You can check RTD builds after pushing PR updates but you can also execute something like `tox r -qq -e build-docs` and check it locally even sooner.

Additionally, spell known abbreviations with capital letters as sequences like "vm" aren't actual words. Though, in most cases it's best to avoid abbreviations and use full words, which reduces cognitive load for less technical readers. Note that RST differs from Markdown: for example, double backticks are used for inline code.

The change log style differs from Git commits since this is a combination of changes that has happened since the previous release, combined in one doc. It's more high-level and often uses something like the past tense or the present simple with the reference to what's changed. While Git commits typically use the imperative mood and focus on the motivation and decisions conserning other developers/contributors but scoped to as smaller change. So in change logs we avoid jargon and make them presentable to a wider audience. In both cases, we should avoid being vague about things and be specific.

As for the byline — we either add it at the end of the sentence and then add a period right after or we put it on a separate lines (with a separator in RST) and don't add a full stop since it's more standalone and isn't a sentence.

Added "macos-15-intel" as vm runner as "macos-13" is not being supported anymore

> -- by :user:`komaldesai13`

Reviewer: ```suggestion
Replaced the decommissioned `macos-13` runner VM image with `macos-15-intel` in CI and CD jobs
-- by :user:`komaldesai13`.
```

> -- by :user:`komaldesai13`.

Reviewer: ```suggestion
-- by :user:`komaldesai13`.

The change includes giving the macOS build job more time to complete as the new version is a little slower.
```

> The change includes giving the macOS build job more time to complete as the new version is a little slower -- by :user:`komaldesai13`

Reviewer: This will give the change log a more consistent look
```suggestion
Now that the ``macos-13`` runner VM image has been decommissioned, the CI and CD jobs have been migrated to use ``macos-15-intel`` -- by :user:`komaldesai13`.

This includes giving the macOS build job more time to complete as the new version is a little slower.
```
